File: sparkhub-frontend/src/api/types/backing.ts
/**
 * 创建订单 (Backing) 的请求 DTO
 */
export interface CreateBackingDTO {
  rewardId: number
}

/**
 * 订单 (Backing) 实体
 * (后端创建订单后返回的对象)
 */
export interface Backing {
  id: number
  backerId: number
  projectId: number
  rewardId: number
  backingAmount: number
  status: number // 0:待支付, 1:已支付, 2:已取消
  createTime: string
}

File: sparkhub/src/main/java/com/pot/sparkhub/entity/Backing.java
package com.pot.sparkhub.entity;

import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Backing implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    private Long id;
    private Long backerId;
    private Long projectId;
    private Long rewardId;
    private BigDecimal backingAmount;
    private Integer status;
    private LocalDateTime createTime;
}